hydra/_internal/hydra.py

```diff
@@ -90,13 +91,18 @@ def run(
             run_mode=RunMode.RUN,
         )
 
-        return run_job(
+        callbacks = Callbacks(cfg)
```

Comment:
One downside of re-initializing is that a callback can no longer share state across different events (e.g, a callback can be a callable class that is updating members across events).

This is still okay for now though.
Are we planning a followup for initializing it once before 1.1?